Week1/SantiagoMedina/unit-test/src/test/java/com/globant/bootcamp/spring/unittest/kata/AccountServiceImplTest.java

```diff
@@ -0,0 +1,86 @@
+package com.globant.bootcamp.spring.unittest.kata;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import springbootcamp.unittest.kata.business.AccountServiceImpl;
+import springbootcamp.unittest.kata.data.AccountRepository;
+import springbootcamp.unittest.kata.domain.Account;
+import springbootcamp.unittest.kata.domain.AccountType;
+import java.lang.reflect.Constructor;
+import static org.mockito.Mockito.when;
+
+public class AccountServiceImplTest {
+
+    @Mock
+    private AccountRepository accountRepository;
+
+    @InjectMocks
+    private AccountServiceImpl accountService;
+
+    private Account currentAccount;
+    private Account savingAccount;
+    
+
+    @Before
+    public void setUp(){
+        MockitoAnnotations.initMocks(this);
+
+        try{
+            Constructor<Account> buildAccount = Account.class.getDeclaredConstructor(String.class, int.class, AccountType.class);
+            buildAccount.setAccessible(true);
+            currentAccount = buildAccount.newInstance("Santiago",0,AccountType.CURRENT);
+            savingAccount = buildAccount.newInstance("Felipe",0,AccountType.SAVING);
+
+        }
+        catch (Exception e){
```

Comment:
This is a unit testing, so, you don't need to catch an exception, let it throws, junit is going to handle and print it for you

Reply:
in this case, the constructor to the class Account is private ... i thought to change the constructor of public but maybe is a business rule, so I created a constructor and set the accessible option in true, but this method "setAccessible" maybe throws a SecurityException for this reason is neccesary a Try catch. Is Easier change the constructor to public in the class Account and do a simple constructor new Account ( ... ). You will tell me which is the best way xD.